blockdag/dag.go

```diff
@@ -1651,49 +1657,50 @@ func (dag *BlockDAG) blueBlocksBetween(lowHash, highHash *daghash.Hash, maxEntri
 		if err != nil {
 			return nil, err
 		}
-		if !isCurrentAncestorOfLowNode {
-			highNodeAntiFuture.Push(current)
-			for _, parent := range current.parents {
-				queue.Push(parent)
-			}
+		if isCurrentAncestorOfLowNode {
+			continue
+		}
+		candidateNodes.Push(current)
+		for _, parent := range current.parents {
+			queue.Push(parent)
 		}
 	}
 
-	// Pop highNodeAntiFuture into a slice. Since highNodeAntiFuture
-	// is an UpHeap, it's guaranteed to be ordered from low to high
+	// Pop candidateNodes into a slice. Since candidateNodes is
+	// an upHeap, it's guaranteed to be ordered from low to high
 	nodesLen := int(maxEntries)
-	if highNodeAntiFuture.Len() < nodesLen {
-		nodesLen = highNodeAntiFuture.Len()
+	if candidateNodes.Len() < nodesLen {
+		nodesLen = candidateNodes.Len()
 	}
 	nodes := make([]*blockNode, 0, nodesLen)
-	for len(nodes) <= nodesLen {
-		nodes = append(nodes, highNodeAntiFuture.pop())
+	for len(nodes) < nodesLen {
```

Comment:
I think it'll be clearer to convert it to `for candidateNodes.Len() > 0 && len(nodes) < maxEntries` and get rid of `nodesLen`